### ISSUE TITLE
Add default accessibility considerations to `DaffProgressBarComponent`

### ISSUE THREAD

**Issue by xelaint** [graycoreio/daffodil#3844]
### Feature description

Add default accessibility considerations to `DaffProgressBarComponent`

`label[daffProgressBarLabel]` is missing a `for` attribute that links to the `id` of the fill div. It currently needs to be manually set by the user, but there's no `id` on the fill div anyway, so it wouldn't really work.

### Use case

As a developer, I would like to provide an accessible experience for screen reader users.

### Daffodil version

0.87.2
